ash-bergs/pantry-pal
# feat: configure webpack

The reasons for these changes are outlined in the issue - https://github.com/ash-bergs/pantry-pal/issues/18 

Basically, we were getting bogged down by all the http requests needed and the blocking time created by all the stylesheet and script tags. 

Webpack has been configured to bundle JS and CSS 

- Adds webpack and webpack.config
- Adds .babelrc 
- Adds `dev`, `build` and `deploy` scripts
- Adds workflow `deploy.yml` to automate the build and deployment process to GitHub Pages


Testing
Local Testing:

- Run npm run dev to start the development server and verify that the project works as expected locally.
- Run npm run build to generate the dist directory and verify the build output.

Deployment Testing:

- Merge this PR to the main branch and verify that the GitHub Actions workflow successfully builds and deploys the project to GitHub Pages.

----

Rollback Plan
If any issues arise after merging, revert the merge commit using:

```bash
git revert -m 1 <merge-commit-hash>
git push origin main
```
